cardano/src/tx.rs

```diff
@@ -354,11 +338,71 @@ impl cbor_event::de::Deserialize for Tx {
     }
 }
 
+/// A transaction witness is a vector of input witnesses
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub struct TxWitness {
+    pub in_witnesses: Vec<TxInWitness>
+}
```

Comment:
Might be worth adding the following instead:

```rust
pub struct TxWitness(Vec<TxInWitness>);
pub Deref for TxWitness {
    type Taget = [TxInWitness];
    fn deref(&self) -> &Self::Target { self.0.deref() }
}
// add the AsRef, IntoIterator, FromIterator traits too
impl From<Vec<TxInWitness>> for TxWitness {
    fn from(witnesses: Vec<TxInWitness>) -> Self { TxWitness(witnesses) }
}
```

On the long run it will simplify our code, and makes the API more user friendly. This will allow construction with the following:

```rust
vec![...].into();
// or
[...].into_iter().collect()
```